procedures/unit-testing-basics.ipf

```diff
@@ -1440,9 +1484,20 @@ static Function/S CheckFunctionSignaturesTC(testCaseList, procWin)
 				sprintf msg, "Data Generator function %s returns a wave with zero points. It is referenced by test case %s.", dgen, fullTestCase
 				UTF_PrintStatusMessage(msg)
 				continue
-			else
-				reducedTCList = AddListItem(testCase, reducedTCList, ";", inf)
+			elseif(!UTF_Utils#IsNaN(wRefSubType) && wType1 == WAVETYPE1_WREF)
+				WAVE/WAVE wWrefType = wGenerator
+				numMD = DimSize(wGenerator, UTF_ROW)
+				for(i1 = 0; i1 < numMD; i1 += 1)
+					WAVE wCheckType = wWrefType[i1]
+					if(WaveType(wCheckType, 1) != wRefSubType)
+						sprintf msg, "Test case %s expects specific wave type1 %u from the Data Generator %s. The wave type at index %u from the data generator does not fit to expected wave type.", fullTestCase, wRefSubType, dgen, i
+						UTF_PrintStatusMessage(msg)
+						Abort msg
+					endif
```

Comment:
How about a sub routine here like

```
static Function HasConstantWaveTypes(wv, subType)
WAVE/WAVE wv
variable subType

Make/FREE/N=(DimSize(wv, UTF_ROWS)) matches = WaveType(wv[0], 1) == wRefSubType

FindValue/V=(0) matches
return V_Value == -1
```

and then

```
if(!HasConstantWaveTypes(wWrefType, wRefSubType))
```

Reply:
ok, utility function is added and used.